sdk/dnsresolver/Azure.ResourceManager.DnsResolver/tests/DnsResolverTestBase.cs

```diff
@@ -66,10 +68,31 @@ protected async Task CreateVirtualNetworkAsync(string virtualNetworkName)
             vnet.AddressPrefixes.Add("10.0.0.0/8");
 
             var subscription = await Client.GetSubscriptions().GetAsync(TestEnvironment.SubscriptionId);
-            var resourceGroup = await subscription.Value.GetResourceGroups().GetAsync(TestEnvironment.ResourceGroup);
+            var resourceGroup = await CreateResourceGroupAsync();
+            var vnetName = Recording.GenerateAssetName("dnsResolver-");
 
-            var virtualNetworks = resourceGroup.Value.GetVirtualNetworks();
-            await virtualNetworks.CreateOrUpdateAsync(WaitUntil.Completed, virtualNetworkName, vnet);
+            var virtualNetworks = resourceGroup.GetVirtualNetworks();
+            //await virtualNetworks.CreateOrUpdateAsync(WaitUntil.Completed, virtualNetworkName, vnet);
+            ResourceIdentifier vnetID;
+            ResourceIdentifier subnetID;
+            if (Mode == RecordedTestMode.Playback)
+            {
+                subnetID = SubnetResource.CreateResourceIdentifier(resourceGroup.Id.SubscriptionId, resourceGroup.Id.Name, vnetName, SubnetName);
+                vnetID = VirtualNetworkResource.CreateResourceIdentifier(resourceGroup.Id.SubscriptionId, resourceGroup.Id.Name, vnetName);
+            }
+            else
+            {
+                using (Recording.DisableRecording())
+                {
+                    var vnetResource = await resourceGroup.GetVirtualNetworks().CreateOrUpdateAsync(WaitUntil.Completed, vnetName, vnet);
+                    var subnetCollection = vnetResource.Value.GetSubnets();
```

Comment:
We could remove this since it is not used.

Reply:
fixed